### PR TITLE
fix gosec in golangci.yaml

### PR DESCRIPTION
**Pull request by wlp1153468871**
Signed-off-by: wlp1153468871 <luping.wei@daocloud.io>

#### What type of PR is this?

/kind cleanup

#### Which issue(s) this PR fixes:

part of #107

#### Which issue(s) this PR fixes:

Fixes #

#### Does this PR introduce a user-facing change?

```release-note
NONE
```

### CODE REVIEW

File: .golangci.yaml [hunk 1]
```diff
@@ -15,7 +15,7 @@ linters:
     - gofmt
     - goimports
     - goprintffuncname
-    # - gosec
+    - gosec
```
Comment (wzshiming) [inline .golangci.yaml:18]: Add follow config in `linters-settings`

```
  gosec:
    excludes:
      - G110
      - G304
    config:
      G301: "0750"
      G302: "0640"
      G306: "0640"
```

Revert the G110 and G304

**Review by wzshiming — CHANGES_REQUESTED** [review]
Revert the G110 and G304

File: pkg/config/config.go [hunk 2]
```diff
@@ -215,7 +216,7 @@ func FilterWithoutTypeFromContext[T metav1.Object](ctx context.Context) (out []m
 
 func loadRawConfig(p string) ([]json.RawMessage, error) {
 	var raws []json.RawMessage
-	file, err := os.Open(p)
+	file, err := os.Open(filepath.Clean(p))
```
Comment (wzshiming) [inline pkg/config/config.go:219]: `filepath.Clean` should be no longer needed

File: pkg/kwokctl/runtime/cluster.go [hunk 3]
```diff
@@ -220,9 +221,9 @@ func (c *Cluster) KubectlInCluster(ctx context.Context, stm exec.IOStreams, args
 }
 
 func (c *Cluster) AuditLogs(ctx context.Context, out io.Writer) error {
-	logs := c.GetLogPath(AuditLogName)
+	logs := filepath.Clean(c.GetLogPath(AuditLogName))
 
-	f, err := os.OpenFile(logs, os.O_RDONLY, 0644)
+	f, err := os.OpenFile(logs, os.O_RDONLY, 0600)
```
Comment (wzshiming) [inline pkg/kwokctl/runtime/cluster.go:226]:
```suggestion
	f, err := os.OpenFile(logs, os.O_RDONLY, 0640)
```

The same goes for everything else

File: pkg/utils/file/untar.go [hunk 4]
```diff
@@ -169,14 +165,24 @@ func untargz(ctx context.Context, src string, filter func(file string) (string,
 				}
 			}()
 
-			if _, err := io.Copy(outFile, tr); err != nil {
-				return err
-			}
-			return nil
+			return copy(outFile, tr)
```
Comment (wzshiming) [inline pkg/utils/file/untar.go:168]: revert this

File: pkg/kwokctl/runtime/binary/cluster.go [hunk 5]
```diff
@@ -592,9 +592,9 @@ func (c *Cluster) Logs(ctx context.Context, name string, out io.Writer) error {
 
 	logger := log.FromContext(ctx)
 
-	logs := c.GetLogPath(filepath.Base(name) + ".log")
+	logs := filepath.Clean(c.GetLogPath(filepath.Base(name) + ".log"))
```
Comment (wzshiming) [inline pkg/kwokctl/runtime/binary/cluster.go:595]:
```suggestion
	logs := c.GetLogPath(filepath.Base(name) + ".log")
```

File: pkg/kwokctl/runtime/cluster.go [hunk 6]
```diff
@@ -225,9 +226,9 @@ func (c *Cluster) KubectlInCluster(ctx context.Context, stm exec.IOStreams, args
 }
 
 func (c *Cluster) AuditLogs(ctx context.Context, out io.Writer) error {
-	logs := c.GetLogPath(AuditLogName)
+	logs := filepath.Clean(c.GetLogPath(AuditLogName))
```
Comment (wzshiming) [inline pkg/kwokctl/runtime/cluster.go:229]:
```suggestion
	logs := c.GetLogPath(AuditLogName)
```

File: pkg/utils/exec/cmd.go [hunk 7]
```diff
@@ -93,7 +93,7 @@ func ForkExec(ctx context.Context, dir string, name string, arg ...string) error
 
 // ForkExecRestart restarts the process if it is not running.
 func ForkExecRestart(ctx context.Context, dir string, name string) error {
-	cmdlinePath := path.Join(dir, "cmdline", filepath.Base(name))
+	cmdlinePath := filepath.Clean(path.Join(dir, "cmdline", filepath.Base(name)))
```
Comment (wzshiming) [inline pkg/utils/exec/cmd.go:96]:
```suggestion
	cmdlinePath := path.Join(dir, "cmdline", filepath.Base(name))
```

**Comment by wzshiming** [comment]
It looks like this change has caused some of the tests to fail, for the docker-compose implementation, docker and nerdctl

File: pkg/utils/exec/cmd.go [hunk 8]
```diff
@@ -35,7 +35,7 @@ import (
 // ForkExec forks a new process and execs the given command.
 // The process will be terminated when the context is canceled.
 func ForkExec(ctx context.Context, dir string, name string, arg ...string) error {
-	pidPath := path.Join(dir, "pids", filepath.Base(name)+".pid")
+	pidPath := filepath.Clean(path.Join(dir, "pids", filepath.Base(name)+".pid"))
```
Comment (wzshiming) [inline pkg/utils/exec/cmd.go:38]: ditto

File: pkg/utils/exec/cmd.go [hunk 9]
```diff
@@ -49,27 +49,27 @@ func ForkExec(ctx context.Context, dir string, name string, arg ...string) error
...
 		return err
 	}
-	err = os.MkdirAll(filepath.Dir(cmdlinePath), 0755)
+	err = os.MkdirAll(filepath.Dir(cmdlinePath), 0750)
 	if err != nil {
 		return err
 	}
 
-	logFile, err := os.OpenFile(logPath, os.O_WRONLY|os.O_CREATE|os.O_TRUNC, 0644)
+	logFile, err := os.OpenFile(filepath.Clean(logPath), os.O_WRONLY|os.O_CREATE|os.O_TRUNC, 0640)
```
Comment (wzshiming) [inline pkg/utils/exec/cmd.go:65]: ditto

File: pkg/utils/exec/cmd.go [hunk 10]
```diff
@@ -107,7 +107,7 @@ func ForkExecRestart(ctx context.Context, dir string, name string) error {
 
 // ForkExecKill kills the process if it is running.
 func ForkExecKill(ctx context.Context, dir string, name string) error {
-	pidPath := path.Join(dir, "pids", filepath.Base(name)+".pid")
+	pidPath := filepath.Clean(path.Join(dir, "pids", filepath.Base(name)+".pid"))
```
Comment (wzshiming) [inline pkg/utils/exec/cmd.go:110]: ditto

File: pkg/utils/file/file.go [hunk 11]
```diff
@@ -24,11 +24,11 @@ import (
 
 // Create creates a file at path with the given content.
 func Create(name string, perm os.FileMode) error {
-	err := os.MkdirAll(filepath.Dir(name), 0755)
+	err := os.MkdirAll(filepath.Dir(name), 0750)
 	if err != nil {
 		return err
 	}
-	file, err := os.OpenFile(name, os.O_WRONLY|os.O_CREATE|os.O_TRUNC, perm)
+	file, err := os.OpenFile(filepath.Clean(name), os.O_WRONLY|os.O_CREATE|os.O_TRUNC, perm)
```
Comment (wzshiming) [inline pkg/utils/file/file.go:31]: ditto

File: pkg/utils/file/file.go [hunk 12]
```diff
@@ -41,12 +41,12 @@ func Create(name string, perm os.FileMode) error {
...
 // Copy copies a file from src to dst.
 func Copy(oldpath, newpath string) error {
-	err := os.MkdirAll(filepath.Dir(newpath), 0755)
+	err := os.MkdirAll(filepath.Dir(newpath), 0750)
 	if err != nil {
 		return err
 	}
 
-	oldFile, err := os.OpenFile(oldpath, os.O_RDONLY, 0)
+	oldFile, err := os.OpenFile(filepath.Clean(oldpath), os.O_RDONLY, 0)
```
Comment (wzshiming) [inline pkg/utils/file/file.go:49]: ditto

File: pkg/utils/file/file.go [hunk 13]
```diff
@@ -59,7 +59,7 @@ func Copy(oldpath, newpath string) error {
 		return err
 	}
 
-	newFile, err := os.OpenFile(newpath, os.O_WRONLY|os.O_CREATE|os.O_TRUNC, fi.Mode())
+	newFile, err := os.OpenFile(filepath.Clean(newpath), os.O_WRONLY|os.O_CREATE|os.O_TRUNC, fi.Mode())
```
Comment (wzshiming) [inline pkg/utils/file/file.go:62]: ditto

File: pkg/utils/file/untar.go [hunk 14]
```diff
@@ -85,7 +85,7 @@ func unzip(ctx context.Context, src string, filter func(file string) (string, bo
 				}
 			}()
 
-			outFile, err := os.OpenFile(name, os.O_WRONLY|os.O_CREATE|os.O_TRUNC, fi.Mode())
+			outFile, err := os.OpenFile(filepath.Clean(name), os.O_WRONLY|os.O_CREATE|os.O_TRUNC, fi.Mode())
```
Comment (wzshiming) [inline pkg/utils/file/untar.go:88]: ditto

File: pkg/utils/file/untar.go [hunk 15]
```diff
@@ -111,7 +111,7 @@ func unzip(ctx context.Context, src string, filter func(file string) (string, bo
 
 func untargz(ctx context.Context, src string, filter func(file string) (string, bool)) error {
 	logger := log.FromContext(ctx)
-	r, err := os.Open(src)
+	r, err := os.Open(filepath.Clean(src))
```
Comment (wzshiming) [inline pkg/utils/file/untar.go:114]: ditto

File: pkg/utils/file/untar.go [hunk 16]
```diff
@@ -154,11 +154,11 @@ func untargz(ctx context.Context, src string, filter func(file string) (string,
 				return nil
 			}
 
-			err = os.MkdirAll(filepath.Dir(name), 0755)
+			err = os.MkdirAll(filepath.Dir(name), 0750)
 			if err != nil {
 				return err
 			}
-			outFile, err := os.OpenFile(name, os.O_WRONLY|os.O_CREATE|os.O_TRUNC, fs.FileMode(hdr.Mode))
+			outFile, err := os.OpenFile(filepath.Clean(name), os.O_WRONLY|os.O_CREATE|os.O_TRUNC, fs.FileMode(hdr.Mode))
```
Comment (wzshiming) [inline pkg/utils/file/untar.go:161]: ditto

**Review by wzshiming — CHANGES_REQUESTED** [review]

File: pkg/utils/file/untar.go (hunk 15 above)
Comment (wlp1153468871) [reply pkg/utils/file/untar.go]: I have solved it，thanks.

File: hack/verify-boilerplate.go [hunk 17]
```diff
@@ -145,7 +146,7 @@ func verifyFile(filePath string) error {
 	}
 
 	// read the file
-	b, err := os.ReadFile(filePath)
+	b, err := os.ReadFile(filepath.Clean(filePath))
```
Comment (wzshiming) [inline hack/verify-boilerplate.go:149]:
```suggestion
	b, err := os.ReadFile(filePath)
```

File: pkg/cni/netns_linux.go [hunk 18]
```diff
@@ -101,7 +102,7 @@ func NewNS(name string) (ns.NetNS, error) {
 	}
 
 	// create an empty file at the mount point
-	mountPointFd, err := os.Create(nsPath)
+	mountPointFd, err := os.Create(filepath.Clean(nsPath))
```
Comment (wzshiming) [inline pkg/cni/netns_linux.go:105]:
```suggestion
	mountPointFd, err := os.Create(nsPath)
```

**Review by wzshiming — CHANGES_REQUESTED** [review]

File: pkg/cni/netns_linux.go [hunk 19]
```diff
@@ -21,6 +21,7 @@ import (
 	"fmt"
 	"os"
 	"path"
+	"path/filepath"
```
Comment (wzshiming) [inline pkg/cni/netns_linux.go:24]:
```suggestion
```

**Review by wzshiming — CHANGES_REQUESTED** [review]

**Review by wzshiming — CHANGES_REQUESTED** [review]

File: pkg/kwokctl/runtime/compose/cluster.go [hunk 20]
```diff
@@ -297,6 +297,10 @@ func (c *Cluster) Install(ctx context.Context) error {
 			return fmt.Errorf("failed to generate prometheus yaml: %w", err)
 		}
 		prometheusConfigPath := c.GetWorkdirPath(runtime.Prometheus)
+
```
Comment (wzshiming) [inline pkg/kwokctl/runtime/compose/cluster.go:300]: Run `gofmt -w ./pkg/kwokctl/runtime/compose/cluster.go` to format this file

**Review by wzshiming — APPROVED** [review]
/approve
/lgtm

**Pull request merged**